server/src/main/java/org/opensearch/index/translog/RemoteFsTimestampAwareTranslog.java

```diff
@@ -312,6 +329,22 @@ protected static List<String> getMetadataFilesToBeDeleted(
             metadataFilesToBeDeleted.size()
         );
 
+        if (indexDeleted == false) {
+            // Filter out metadata files based on minGenerationToKeep
+            List<String> metadataFilesContainingMinGenerationToKeep = metadataFilesToBeDeleted.stream().filter(md -> {
+                long maxGeneration = TranslogTransferMetadata.getMaxGenerationFromFileName(md);
+                return maxGeneration == -1 || maxGeneration >= minGenerationToKeepInRemote;
```

Comment:
why are we checking maxGeneration instead of minGeneration ?

Reply:
We want to keep metadata files which contain at least `minGenerationToKeepInRemote`. 

If `minGenerationToKeepInRemote = 120`, we don't want to delete metadata files which have following min-max generation range:
- 100-140
- 80-120
- 110-160

But we can safely delete files which have following min-manx generation range:
- 50-80
- 100-115
- 110-119